### ISSUE TITLE
Non editable project 

### ISSUE THREAD

**Issue by SalahedeenIssa** [Infleqtion/client-superstaq#714]
### What is happening?

The changes I was making locally to cirq_superstaq were not being reflected when I was testing them locally. This turns out to be because I installed it using an outdated version of pip (<23). When I installed it again using a version of pip that was >23, it seemed to work fine. 

### How can we reproduce the issue?

Install css or qss using pip<23

### What should happen?

You should be able to see local changes on local branches reflected when running locally. 

### Environment

Pip < 23

**Comment by perlinm** [comment]
@richrines1 @vtomole my first instinct was to add `pip>=23.0.0` to `gss/dev-requirements.txt`, but I'm not sure that would work.  This seems to be a bit of a chicken-and-egg problem.

I can't seem to reproduce the issue locally, since if I make a fresh environment and install `pip==22.0.0`, I can still `pip install -e .` just fine inside `css`.

@SalahedeenIssa can you reproduce this issue in a new environment?  If so, can you check whether adding `pip>=23.0.0` to one of `general-superstaq/dev-requirements.txt` fixes the issue?

**Comment by SalahedeenIssa** [comment]
@richrines1 That didn't seem to fix it - I still see cirq_superstaq as appearing only in the site packages when I do this

**Comment by perlinm** [comment]
If adding `pip>=23.0.0` works, then let's just add that to `gss/dev-requirements.txt`.

**Comment by perlinm** [comment]
@SalahedeenIssa just to clarify: is it the case that
- (a) making a fresh environment with pip<23, and then (b) running `pip install -e .` does not work, but
- doing the same as above but with `pip>=23.0.0` in `gss/dev-requirements.txt` does work?

I'm trying to make sure that `pip` was not upgraded manually in this process.

**Comment by SalahedeenIssa** [comment]
Yes that was the case for me

**Comment by perlinm** [comment]
Actually, this doesn't make sense.  It would maybe make sense if adding `pip>=23.0.0` to `requirements.txt` fixed `pip install -e .`, but that command shouldn't touch `dev-requirements.txt`.  Let's try to reproduce this issue live/in-person @SalahedeenIssa to figure it out.

**Comment by SalahedeenIssa** [comment]
Oh in reality I don't even think I added pip>=23.0.0 to any requirements file, but I just manually upgraded it. I thought that since those are effectively the same it would make sense for me to say that it was the case for me

**Comment by perlinm** [comment]
No, I'm distinguishing you installing pip manually (not what we want), vs. pip figuring out on its own that it needs to upgrade itself (which would fix the issue in this thread).
